CRM/Contribute/BAO/Contribution.php

```diff
@@ -141,14 +140,19 @@ public static function add(&$params, $ids = []) {
         $params['creditnote_id'] = self::createCreditNoteId();
       }
     }
-    if (empty($params['contribution_status_id'])) {
+    $contributionStatusID = $params['contribution_status_id'] ?? NULL;
+    if (CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', (int) $contributionStatusID) === 'Partially paid' && empty($params['is_post_payment_create'])) {
+      CRM_Core_Error::deprecatedFunctionWarning('Setting status to partially paid other than by using Payment.create is deprecated and unreliable');
+    }
+    if (!$contributionStatusID) {
```

Comment:
@JoeMurray So if $params['contribution_status_id'] was set to NULL contribution_status_id key was not included then we should get the contribution status Id here